design-proposals/passt-feature-gate-iface-api-deprecation.md

```diff
@@ -0,0 +1,61 @@
+## Overview
+Following the emerging [Kubevirt Network Binding Plugin API](https://docs.google.com/document/d/13fLaqZuM8El_CT5XzxPv-1epKsB1-Spv_jtXveEkkdQ/edit#heading=h.z27rudfpwy44)
+and introduction of Passt network binding plugin [2](https://github.com/kubevirt/kubevirt/pull/10425) [3](https://github.com/kubevirt/kubevirt/pull/10621), 
+it's now possible to remove Passt network binding implementation from Kubevirt core.
+
+This doc outlines deprecation plan for Passt network binding, interface API and  feature gate.
+
+# Motivation
+Removing Passt network binding implementation from Kubevirt core and use Passt network binding plugin instead, 
+enable reduce the maintenance burden.
+Having Passt binding code outside Kubevirt's core may ease contribution process as there is no need to be familiar 
+with [kubevirt/kubevirt](https://github.com/kubevirt/kubevirt) core code.
+
+## Goals
+- Deprecate Passt network interface API.
+- Remove Passt feature gate.
+- Remove Passt network binding from Kubevirt core.
+  
+## Non Goals
+
+## User Stories
+
+## Repos
+[kubevirt/kubevirt](https://github.com/kubevirt/kubevirt):
+- [Passt binding plugin hook sidecar](https://github.com/kubevirt/kubevirt/tree/main/cmd/sidecars/network-passt-binding)
+- [Passt binding plugin CNI](https://github.com/kubevirt/kubevirt/tree/main/cmd/cniplugins/passt-binding)
+
+# Design
+Deprecating Passt feature-gate and interface API and replacing it with Passt binding plugin
+require some changes users should be aware of and prepared for.
+
+The proposed deprecation plan for Passt binding interface API and feature-gate is as follows:
+
+1. Announce passt feature-gate and interface API deprecation in kubevirt-dev mailing list.
+   This will give users some time to prepare for changes in Kubevirt next release v1.2.0. 
+
+2. Kubevirt next release v1.2.0 should have the following changes: 
+   1. [Passt interface API](http://kubevirt.io/api-reference/main/definitions.html#_v1_interfacepasst) 
+      usage shall be blocked by the admission webhook with a message saying binding plugin API should be
+      used instead.
+      
+      To user Passt binding the user will have to follow the network binding plugin API:
+      1. Install Passt binding CNI.
+      2. Create and register Passt binding network-attachment-definition.
+      3. Register Passt binding hook sidecar image.
```

Comment:
We do not call it "hook" on the API.

Reply:
In general this what installing a CNI means.

I dont mind being explicit about it and mention cluster nodes, done.

Reply:
Done